### ISSUE TITLE
Not Resuling Correct Output for Full Adder Thruth Table

### ISSUE THREAD

**Issue by ridwan257** [CodingTrain/Toy-Neural-Network-JS#168]
I had trained with this data which is truth table of full adder,
`brain = new NeuralNetwork(3, 12, 2);
  inputs = [
    [0, 0, 0],
    [0, 0, 1],
    [0, 1, 0],
    [0, 1, 1],
    [1, 0, 0],
    [1, 0, 1],
    [1, 1, 0],
    [1, 1, 1]
]

outputs = [
    [0, 0],
    [1, 0],
    [1, 0],
    [0, 1],
    [1, 0],
    [0, 1],
    [0, 1],
    [1, 1]
]`

after 50K iteration with training data, this model only giving correct output for [0, 0, 0], [0, 0, 1], [0, 1, 0], [0, 1, 1], rest others not showing correct prediction. I have both tried sigmoid and tanh function. but no improvement. I also tried many combinations of hidden layer, learning rate. but doest show improvement!
